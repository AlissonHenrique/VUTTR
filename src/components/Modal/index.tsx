import ReactModal from "react-modal";
import { ReactNode, useEffect, useState } from "react";
interface PropsModal {
  isOpen: boolean;
  children: ReactNode;
  setIsOpen: () => void;
}

export function Modal({ children, isOpen, setIsOpen }: PropsModal) {
  const [modalStatus, setModalStatus] = useState(isOpen);
  useEffect(() => {
    setModalStatus(isOpen);
  }, [isOpen]);

  return (
    <ReactModal
      shouldCloseOnOverlayClick={!false}
      onRequestClose={setIsOpen}
      isOpen={modalStatus}
      overlayClassName="react-modal-overlay"
      className="react-modal-content"
    >
      {children}
    </ReactModal>
  );
}
