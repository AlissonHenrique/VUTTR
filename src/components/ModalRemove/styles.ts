import styled from "styled-components";

export const BgModal = styled.div`
  width: 100%;
  height: 100%;
  background: #170c3ae6 0% 0% no-repeat padding-box;
  position: absolute;
  top: 0;
  bottom: 0;
  z-index: 10;
  overflow-y: hidden;
  @media (max-width: 1920px) {
    height: 1090px;
  }
  .container-modal {
    max-width: 600px;

    background: #ffffff 0% 0% no-repeat;
    box-shadow: 0px 20px 25px #0000001a;
    border-radius: 5px;
    margin: 10% auto 0 auto;
    padding: 30px;

    p {
      margin: 30px 0;
      color: #8f8a9b;
      font-size: 20px;
    }
  }
  .header-modal {
    display: flex;
    align-items: center;
    justify-content: space-between;
    h4 {
      font-size: 26px;
    }
    svg {
      cursor: pointer;
    }
  }
`;
