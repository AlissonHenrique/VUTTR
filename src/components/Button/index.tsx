import React from "react";

import { Container } from "./styles";
interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  colorType: string;
}
function Button({ children, colorType, ...rest }: Props) {
  return (
    <Container {...rest} className={colorType}>
      {children}
    </Container>
  );
}

export default Button;
