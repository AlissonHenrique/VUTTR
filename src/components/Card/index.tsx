import Button from "../Button";
import { Container } from "./styles";

interface Tools {
  id: number;
  title: string;
  link: string;
  description: string;
  tags: string[];
}
interface PropsCard {
  handleOpenModalRemove: (id: number) => void;
  tool: Tools;
}

export function Card({ handleOpenModalRemove, tool }: PropsCard) {
  return (
    <Container>
      <div className="card-header">
        <a href={tool.link} rel="noreferrer" target="_blank">
          {tool.title}
        </a>
        <Button colorType="red" onClick={() => handleOpenModalRemove(tool.id)}>
          Remove
        </Button>
      </div>
      <p>{tool.description}</p>
      <span>{tool.tags}</span>
    </Container>
  );
}
