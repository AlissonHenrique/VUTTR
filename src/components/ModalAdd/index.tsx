import { GrFormClose } from "react-icons/gr";
import Button from "../Button";
import { Modal } from "../Modal";
import { useCallback, useState } from "react";

import * as Yup from "yup";
import { toast } from "react-toastify";

interface PropsItems {
  id: number;
  title: string;
  link: string;
  description: string;
  tags: string[];
}

interface PropsModalAdd {
  isOpen: boolean;
  setIsOpen: () => void;
  handleSubmit: (items: PropsItems) => void;
}

export function ModalAdd({ isOpen, setIsOpen, handleSubmit }: PropsModalAdd) {
  const [title, setTitle] = useState("");
  const [link, setLink] = useState("");
  const [description, setDescription] = useState("");
  const [tags, setTags] = useState("");

  const handleAddSubmit = useCallback(
    async (e) => {
      e.preventDefault();

      try {
        const schema = Yup.object().shape({
          title: Yup.string().required("Title is required"),
          link: Yup.string().required("Link is required"),
          description: Yup.string().required("Description is required"),
          tags: Yup.array().required("Tags is required"),
        });

        const data = {
          id: Math.floor(Math.random() * (1000 - 1)) + 1,
          title,
          link,
          description,
          tags: tags.split(" ").map((t: string) => `#${t} `),
        };

        await schema.validate(data, {
          abortEarly: false,
        });

        handleSubmit(data);
        setTitle("");
        setLink("");
        setDescription("");
        setTags("");
        setIsOpen();
        toast.success("Tool registred");
      } catch (err) {
        if (err instanceof Yup.ValidationError) {
          // const errors = JSON.stringify(err.errors);
          for (let index in err.inner) {
            toast.error(err.inner[index].message);
          }
        }
      }
    },
    [handleSubmit, setIsOpen, title, link, description, tags]
  );
  return (
    <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
      <div className="header-modal">
        <h4>Add new tool</h4>
        <GrFormClose size={20} color="#8F8A9B" onClick={setIsOpen} />
      </div>
      <form onSubmit={handleAddSubmit}>
        <div className="content-inpt">
          <label>Tool Name</label>
          <input
            type="text"
            value={title}
            onChange={(e) => setTitle(e.target.value)}
          />
        </div>
        <div className="content-inpt">
          <label>Tool Link</label>
          <input
            type="text"
            value={link}
            onChange={(e) => setLink(e.target.value)}
          />
        </div>
        <div className="content-inpt">
          <label>Tool description</label>
          <textarea
            value={description}
            onChange={(e) => setDescription(e.target.value)}
          />
        </div>
        <div className="content-inpt">
          <label>Tags</label>
          <input
            type="text"
            value={tags}
            onChange={(e) => setTags(e.target.value)}
          />
        </div>
        <div className="content-buttons">
          <Button type="submit" colorType="blue-2">
            Add tool
          </Button>
        </div>
      </form>
    </Modal>
  );
}
