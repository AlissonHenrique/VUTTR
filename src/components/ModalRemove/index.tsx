import React from "react";
import { GrFormClose } from "react-icons/gr";
import Button from "../Button";
import { Modal } from "../Modal";

interface PropsModalRemove {
  isOpen: boolean;
  setIsOpenRemove: () => void;
  handleRemove: () => void;
}

export function ModalRemove({
  isOpen,
  setIsOpenRemove,
  handleRemove,
}: PropsModalRemove) {
  return (
    <Modal isOpen={isOpen} setIsOpen={setIsOpenRemove}>
      <div className="header-modal">
        <h4>Remove tool</h4>
        <GrFormClose size={20} color="#8F8A9B" onClick={setIsOpenRemove} />
      </div>
      <p>
        Are you sutre you want to remove <b>hotel?</b>
      </p>
      <div className="content-buttons">
        <Button colorType="red-2" onClick={setIsOpenRemove}>
          Cancel
        </Button>
        <Button colorType="blue-2" onClick={handleRemove}>
          Yes, remove
        </Button>
      </div>
    </Modal>
  );
}
