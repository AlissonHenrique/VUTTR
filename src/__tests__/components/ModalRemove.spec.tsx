import { ModalRemove } from "../../components/ModalRemove";
import { render, fireEvent } from "@testing-library/react";
describe("Modal Remove", () => {
  it("should be able to open modal remove", async () => {
    const mockToggleModalRemove = jest.fn();
    const mockHandleRemove = jest.fn();
    const mockModalOpenRemove = true;

    const modal = render(
      <ModalRemove
        isOpen={mockModalOpenRemove}
        setIsOpenRemove={mockToggleModalRemove}
        handleRemove={mockHandleRemove}
      />
    );

    expect(modal).toMatchSnapshot();
  });
});
