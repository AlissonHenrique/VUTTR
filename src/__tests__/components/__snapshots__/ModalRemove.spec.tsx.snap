// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Remove should be able to open modal remove 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div />
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    >
      <div
        class="ReactModal__Overlay ReactModal__Overlay--after-open react-modal-overlay"
      >
        <div
          aria-modal="true"
          class="ReactModal__Content ReactModal__Content--after-open react-modal-content"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="header-modal"
          >
            <h4>
              Remove tool
            </h4>
            <svg
              color="#8F8A9B"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(143, 138, 155);"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7,7 L17,17 M7,17 L17,7"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <p>
            Are you sutre you want to remove 
            <b>
              hotel?
            </b>
          </p>
          <div
            class="content-buttons"
          >
            <button
              class="sc-bdfBwQ lntmAL red-2"
            >
              Cancel
            </button>
            <button
              class="sc-bdfBwQ lntmAL blue-2"
            >
              Yes, remove
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
